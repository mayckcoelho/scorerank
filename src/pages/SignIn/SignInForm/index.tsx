import React, { useRef } from 'react';
import { Form } from '@unform/mobile';
import { SubmitHandler, FormHandles } from '@unform/core';
import * as Yup from "yup";

import { useAuth } from '../../../contexts/auth';
import Input from "../../../components/Input";
import Buttom from "../../../components/Buttom";
import ISignInRequest from "../../../interfaces/ISignInRequest"
import {
    ForgotPassword,
    ForgotPasswordText,
} from './styles'
interface IDictionary {
    [index: string]: string;
}

const SignInForm: React.FC = () => {
    const { signIn } = useAuth();
    const formRef = useRef<FormHandles>(null);

    const handleSubmit: SubmitHandler<ISignInRequest> = async (data) => {
        try {
            const schema = Yup.object().shape({
                email: Yup.string().email("Informe um e-mail válido").required("E-mail obrigatório"),
                password: Yup.string().min(6, "Informe ao menos 6 caractéres").required("Senha obrigatória"),
            });

            await schema.validate(data, {
                abortEarly: false
            });

            signIn(data);
        } catch (err) {
            const validationErrors = {} as IDictionary;
            if (err instanceof Yup.ValidationError) {
                err.inner.forEach(error => {
                    validationErrors[error.path] = error.message;
                });
                formRef.current?.setErrors(validationErrors);
            }
        }
    };

    return (
        <Form ref={formRef} onSubmit={handleSubmit}>
            <Input name="email"
                returnKeyType="next"
                onSubmitEditing={() => {
                    const passwordInput = formRef.current?.getFieldRef('password');
                    passwordInput.focus();
                }}
                blurOnSubmit={false}
                icon="envelope-o"
                placeholder="Endereço de Email" />
            <Input name="password" icon="key" placeholder="Senha" secureTextEntry />
            <ForgotPassword>
                <ForgotPasswordText>Esqueceu sua senha?</ForgotPasswordText>
            </ForgotPassword>
            <Buttom text="Continuar" color="#215760" onPress={() => formRef.current?.submitForm()} />
        </Form>
    )
}

export default SignInForm;