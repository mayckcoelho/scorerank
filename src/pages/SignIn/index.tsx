import React from 'react'
import {
    View,
    Platform,
    Text,
    TouchableOpacity
} from 'react-native'

import SignInForm from "./SignInForm";
import FacebookButtom from "../../components/FacebookButtom";
import GoogleButtom from "../../components/GoogleButtom";
import {
    Title,
    KeyboardAvoiding,
    LogoImage,
    RegisterView,
    RegisterText
} from './styles'

interface Props {
    navigation?: any;
}

const SignIn: React.FC<Props> = ({ navigation }) => {

    return (
        <KeyboardAvoiding behavior={Platform.OS === 'ios' ? 'padding' : undefined}>
            <View>
                <LogoImage source={require("../../assets/logo.png")} resizeMode="contain" />
                <Title>Entre na sua conta</Title>
                <SignInForm />
                <RegisterView>
                    <Text>NÃ£o tem uma conta?</Text>
                    <TouchableOpacity onPress={() => navigation.navigate("Register")}>
                        <RegisterText>
                            Crie uma aqui
                        </RegisterText>
                    </TouchableOpacity>
                </RegisterView>
                <FacebookButtom />
                <GoogleButtom />
            </View>
        </KeyboardAvoiding>
    )
};

export default SignIn;