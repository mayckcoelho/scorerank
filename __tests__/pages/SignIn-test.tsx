import React from 'react';
import SignIn from '../../src/pages/SignIn';
import renderer from 'react-test-renderer';

describe('SignIn should render correctly', () => {
    it('match snapshot', () => {
        const signin = renderer
            .create(<SignIn />)
            .toJSON();
        expect(signin).toMatchSnapshot();
    });
})