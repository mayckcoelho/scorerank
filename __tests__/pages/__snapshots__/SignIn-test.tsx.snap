// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render correctly match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#f6f7fb",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/logo.png",
        }
      }
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 120,
            "marginBottom": 40,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 40,
            "width": 180,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#455675",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 15,
          },
        ]
      }
    >
      Entre na sua conta
    </Text>
    <View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 15,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 4,
                "color": "#444",
                "flexDirection": "row",
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#b5c0ce",
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={false}
            keyboardAppearance="dark"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Endereço de Email"
            placeholderTextColor="#b5c0ce"
            rejectResponderTermination={true}
            returnKeyType="next"
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 15,
                  "marginLeft": 5,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 15,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 4,
                "color": "#444",
                "flexDirection": "row",
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#b5c0ce",
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <TextInput
            allowFontScaling={true}
            keyboardAppearance="dark"
            onChangeText={[Function]}
            placeholder="Senha"
            placeholderTextColor="#b5c0ce"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontSize": 15,
                  "marginLeft": 5,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#455675",
              },
            ]
          }
        >
          Esqueceu sua senha?
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#215760",
            "borderColor": "black",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 40,
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 15,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Continuar
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "flexDirection": "row",
            "marginBottom": 30,
          },
        ]
      }
    >
      <Text>
        Não tem uma conta?
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#215760",
                "marginLeft": 5,
              },
            ]
          }
        >
          Crie uma aqui
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3b5998",
          "borderColor": "black",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 16,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginLeft": 10,
            },
          ]
        }
      >
        Continuar com Facebook
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#edeef2",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ea4335",
              "fontSize": 16,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#455675",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginLeft": 10,
            },
          ]
        }
      >
        Continuar com Google
      </Text>
    </View>
  </View>
</View>
`;
